src/main/java/graphql/servlet/GraphQLServlet.java

```diff
@@ -266,14 +286,42 @@ private GraphQL newGraphQL(GraphQLSchema schema) {
             .build();
     }
 
-    private void query(String query, String operationName, Map<String, Object> variables, GraphQLSchema schema, HttpServletRequest req, HttpServletResponse resp, GraphQLContext context, Object rootObject) throws IOException {
+    private void doQuery(GraphQLRequest graphQLRequest, GraphQLSchema schema, GraphQLContext context, Object rootObject, HttpServletRequest httpReq, HttpServletResponse httpRes) throws Exception {
+        doQuery(graphQLRequest.getQuery(), graphQLRequest.getOperationName(), graphQLRequest.getVariables(), schema, context, rootObject, httpReq, httpRes);
+    }
+
+    private void doQuery(String query, String operationName, Map<String, Object> variables, GraphQLSchema schema, GraphQLContext context, Object rootObject, HttpServletRequest req, HttpServletResponse resp) throws Exception {
+        query(query, operationName, variables, schema, context, rootObject, (r) -> {
+            resp.setContentType(APPLICATION_JSON_UTF8);
+            resp.setStatus(r.getStatus());
+            resp.getWriter().write(r.getResponse());
+        });
+    }
+
+    private void doBatchedQuery(Iterator<GraphQLRequest> graphQLRequests, GraphQLSchema schema, GraphQLContext context, Object rootObject, HttpServletRequest req, HttpServletResponse resp) throws Exception {
+        resp.setContentType(APPLICATION_JSON_UTF8);
+        resp.setStatus(STATUS_OK);
```

Comment:
A call to `query` below might result in an exception. Is it ok to set this here?